### ISSUE TITLE
Better fps with KCauldron (I want to embrace Thermos)

### ISSUE THREAD

**Issue by chezpaul** [TCPR/Thermos#297]
**Server Log:** http://pastebin.com/5jUWQxA5

**FML Log(s):** None

**Explanation of issue:**
I compared FPS’s on my Kcaulron 1614.188 with my Thermos 1614.55 (same exact server)
Kcaulron 1614.188 is 70 fps
Thermos 1614.55  is 19 fps.

Whatever I do, wherever I go, with same initialization command line.
Tried yours and mine, it doesn’t change the fps.

**How to recreate this issue**:
Stop server, change files, relaunch server, both ways. Always the same results.
Modpack Name: Private

**Mods Installed:**
appliedenergistics2-core, Aroma1997Core, CodeChickenCore, Micdoodlecore, NotEnoughItems, PlayerAPI, SmartCore, OpenModsCore, <CoFH ASM>, arsmagica2, lucky, securitycraft, Treecapitator, additionalpipes, AdvancedSolarPanel, AFSU, AnimationAPI, appliedenergistics2, Aroma1997CoreHelper, bdlib, BetterChests, BiblioCraft, bigtrees, BiomesOPlenty, bookshelf, Botania, BuildCraft|Builders, BuildCraft|Transport, BuildCraft|Energy, BuildCraft|Silicon, BuildCraft|Robotics, BuildCraft|Core, BuildCraft|Factory, BuildCraft|Compat, CarpentersBlocks, ChickenChunks, chisel, chocolateQuest, CoFHCore, colourfulportalsmod, ComputerCraft, ComputerCraftEdu, BuildMod, CoroPets, CoroAI, ExtendedRenderer, ConfigMod, craftguide, CustomOreGen, PTRModelLib, props, MoCreatures, Dynmap, emeraldmod, eplus, endercore, EnderIO, EnderStorage, EnderZoo, etfuturum, exoticbirds, extracells, ExtraUtilities, FinderCompass, Forestry, GalacticraftMars, GalacticraftCore, GalaxySpace, GardenContainers, GardenCore, GardenStuff, GardenTrees, GraviGun, IC2NuclearControl, iChunUtil, imc, IC2, IronChest, ironchestminecarts, ISM-Server, Pads, JustAnotherSpawner, lom, LunatriusCore, magicalcrops, malisiscore, malisisdoors, Mantle, Mekanism, MekanismGenerators, MekanismTools, MineChess, mfs, mfrmagiccropscompat, MineFactoryReloaded, MineFactoryReloaded|CompatAppliedEnergistics, MineFactoryReloaded|CompatBuildCraft, MineFactoryReloaded|CompatForestry, MineFactoryReloaded|CompatForgeMicroblock, MineFactoryReloaded|CompatIC2, MineFactoryReloaded|CompatRailcraft, MineFactoryReloaded|CompatThaumcraft, MineFactoryReloaded|CompatThermalExpansion, MineFactoryReloaded|CompatTConstruct, MineFactoryReloaded|CompatTwilightForest, MineFactoryReloaded|CompatVanilla, powersuits, Morpheus, cfm, Natura, numina, Oceancraft, OpenBlocks, OCS, OpenMods, getalltheseeds, harvestcraft, parachutemod, PowerConverters, pressure, ProjRed|Core, ProjRed|Compatibility, ProjRed|Integration, ProjRed|Transmission, ProjRed|Illumination, Railcraft, RandomThings, RedstoneArsenal, row, rowam, Schematica, secretroomsmod, SGCraft, simplefluidtanks, SmartMoving, SmartRender, SolarFlux, SophisticatedWolves, SpecialMobs, StevesCarts, StorageDrawers, TConstruct, Thaumcraft, ThermalDynamics, ThermalExpansion, ThermalFoundation, recycling, toolsdoneright, Translocator, tropicraft, TwilightForest, Waila, wailaplugins, weather2, witchery, WR-CBE|Core, WR-CBE|Addons, WR-CBE|Logic, bspkrsCore, Baubles, eureka, ForgeMicroblock, ForgeMultipart, McMultipart, MrTJPCoreMod]

**Plugins Installed:**
Worldedit, Worldguard, Vault, Dynmap, LWC, Group Manager, MobArenas, ClearLag, Essentials, Multiverse Core, Portal, Netherportal, Inventories.Prism, Server Signs, ProtocolLib, JsonAPI

**Warmroast Report**: N/A

**Thermos Version:**
55
**Forge Version:**
1614

**Comment by gafunbird007** [comment]
Isn't b55 is bleeding?

**Comment by DiggyWig** [comment]
This has been the case since at least Build 46 (earliest release I tried it with), from my experience that mod actually drastically decreases performance on Thermos, I stopped using it shortly after switching from KCauldron.

**Comment by Robotia** [comment]
O.o
So this is client side related to FPS? By any chance is there weather effects at the time?

**Comment by DiggyWig** [comment]
Yes it is and no it's a general performance decrease regardless of weather. I'm not sure exactly how the mod works but apparently it alters sine and cosine calculations which according to the mod author can improve chunk generation, entity acceleration and other functions when used on a server.

I got mixed results with KCauldron (it was often hard to tell if it was making a real difference or not) so I wasn't bothered about loosing it, it can work wonders when it's used client-side though, which I still do.

**Comment by iKaneki-Ken** [comment]
Server side software does not affect clientside FPS. Never has, never will.

**Comment by chezpaul** [comment]
Guys, this is not a discussion on the Better FPS mod. I don't use it.
This is about my server running the same plugins and mods (which are listed above) on both KCauldron and Thermos. I get more fps (way more) with KCauldron than with Thermos.
I just wanted to put it out there.
I really want to use Thermos. I can see many more quicker things happening. Like elevators are way quicker....
Sorry about the confusion.

**Comment by chezpaul** [comment]
And this is client side fps, yes.
And I do have Weather 2 mod.

**Comment by Robotia** [comment]
What's the fps on Forge?

**Comment by chezpaul** [comment]
I don't know. Sorry. Would have to re build it. Last forge I had was 1492.

**Comment by djoveryde** [comment]
Correct me if I'm wrong but serverside can only affect the TPS of the player not FPS as FPS relates to how fast your system can physically load the graphics which is all loaded by the client side area. All the server does is relays chunks to the player and other things which all relate to ticks?

**Comment by djoveryde** [comment]
Run a single player instance as that will be forge.

**Comment by Robotia** [comment]
If the single player instance has lower fps than KCauldron then it is likely the weather packet fix that I did to fix the weather effects not showing properly.

**Comment by DiggyWig** [comment]
Oops, sorry that's my bad on the Better FPS thing chezpaul. I was having a conversation with someone about the mod in relation to Thermos/KCauldron recently and the original title led me to think it was the same issue.

**Comment by ChenGang0415** [comment]
PlayerDeathEvent.setKeepInventory(true)
this method invalid
clear all item on respawn
please fix it
